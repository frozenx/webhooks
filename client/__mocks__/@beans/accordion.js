

import React from 'react';

const Accordion = (props) => (<div {...props} />);

export default { Accordion };