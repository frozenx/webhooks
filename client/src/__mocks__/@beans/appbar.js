import React from 'react';
const AppBar = (props) => (<div {...props}/>);

export default AppBar;
