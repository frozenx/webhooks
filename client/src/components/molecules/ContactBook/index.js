import React, { Fragment } from "react";
import { FormHeading, PrimaryWrapper, SecondaryWrapper, FormHeadingWrapper } from "../../../common/styles";
import Button from '@beans/button';
import BackComponent from '../SupplierForm/BackComponent'
import CardListWithPagination from '../ContactCardList'
import i18Data from '../../../languagepack';

const ContactBook = ({
    showContactForm,
    contactList,
    getNewPage,
    partnerName,
    handleBackComponent,
    navigateToForm,
    showLoader,
    totalCount,
    currentPage,
    handlePageChange

}) => {
    return (
        <Fragment>
            <BackComponent handleBackComponent={handleBackComponent} />
            <PrimaryWrapper>
                <SecondaryWrapper>
                    <FormHeadingWrapper>
                        <FormHeading>
                            <h1>
                                {`${partnerName} ${i18Data.contacts}`}
                            </h1>
                        </FormHeading>
                        <Button onClick={showContactForm} variant='secondary'>
                            {i18Data.addNewContact}
                        </Button>
                    </FormHeadingWrapper>
                    <CardListWithPagination 
                        paginationProps = {
                            {
                                totalCount,
                                getNewPage, 
                                linkTemplate : "/page=%{pageNumber}",
                                pageText : "Page",
                                currentPage, 
                                handlePageChange
                            }
                        }
                        componentProps = {
                            {
                                contactList,
                                showLoader,
                                navigateToForm,
                                handleBackComponent

                            }
                        }
                    />
                </SecondaryWrapper>
            </PrimaryWrapper>
        </Fragment>
    );
}


export default ContactBook;