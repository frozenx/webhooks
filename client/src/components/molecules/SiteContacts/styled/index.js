import styled from 'styled-components'

export const DropdownWrapper = styled.div`
    label{
        margin-bottom: 8px;
    }
`
export const StyledContainer = styled.div`
    margin-bottom: 24px;
`
