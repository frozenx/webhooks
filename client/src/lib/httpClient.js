const redirectUrls = require('../config').redirect;


const handleError = (errorStatus) => {
    if (errorStatus === 403) {
        window.location = redirectUrls.unauthorized;
    } else if (errorStatus === 401) {
        window.location = redirectUrls.externalUrl;
    }
    else if (errorStatus === 404) {
        window.location = redirectUrls.pageNotFound;
    }
    else {
        window.location = redirectUrls.serverError;
    }
}

async function get(url) {
    try {
        const response = await fetch(url, {
            method: 'get',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
        });
        if (response.ok) {
            const body = await response.json();
            return body;
        } else {
            throw response;
        }
    }
    catch (err) {
        if (err.status) {
            handleError(err.status);
        }
    }
}

async function post(url, payload) {
    try {
        const response = await fetch(url, {
            method: 'POST',
            body: JSON.stringify(payload),
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
        });
        if (response.ok) {
            const body = await response.json();
            return body;
        } else {
            throw response;
        }
    }
    catch (err) {
        if (err.status) {
            handleError(err.status);
        }
    }
}


export default { get, post };

