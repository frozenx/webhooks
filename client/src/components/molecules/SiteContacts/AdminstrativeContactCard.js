import React, { Fragment } from 'react';
import { StyleGrayWrapper } from "../../../common/styles";
import Label from '@beans/label';
import Button from '@beans/button';
import i18Data from '../../../languagepack';

import {
    CardTopContainer,
    CardBottomContainer,
    StyledLeftSection,
    StyledRightSection,
    IconStyle
} from '../../../common/styles'
import StatusIndicator from '@beans/colleague-status-indicator';
import Dropdown from '@beans/dropdown';
import Icon from '@beans/icon';
import { DropdownWrapper } from './styled'


export default class AdministrativeContactCard extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            administrativeContact: {selectedContacts:[], availableContacts:[]},
        }
    }
    getContact = (uuids) => {                
        let contactIds = uuids.split(',')
        let contacts = []
        contactIds.forEach(contactId => {            
            let foundContact = this.props.rootNode.contactBook.getContact(contactId)
            
            if(foundContact) {
                let { contactStatus, firstName, lastName, contactUuid } = foundContact;
                let contact = { contactStatus, firstName, lastName, contactUuid }
                contacts.push(contact)
            }
        })
            return contacts;
        }

    componentDidMount() {        
        let savedAdministrativeContacts = this.props.node.contact.values['administrativeContact'] && this.props.node.contact.values['administrativeContact'] || '';
        let administrativeContact = { ...this.state.administrativeContact }
        let isSavedContactTypesEmpty = !savedAdministrativeContacts.length > 0
        let contactsReceived = [...this.props.contacts];
        if (!isSavedContactTypesEmpty) {            
                let foundContacts = this.getContact(savedAdministrativeContacts);                
                administrativeContact.selectedContacts = [...foundContacts]
                let filteredAvailableContacts = contactsReceived.filter(availableContact => {
                    return  foundContacts.every(({contactUuid})=> availableContact.contactUuid !==contactUuid);
                })
                administrativeContact.availableContacts = [...filteredAvailableContacts]
              this.setState({
                    administrativeContact
                })
        }

        else {
            administrativeContact.availableContacts = contactsReceived               
            this.setState({
                administrativeContact,
            })

            }            
        }
    onChange = (e) => {
        let foundContact = this.getContact(e.target.value)
        let administrativeContact = {...this.state.administrativeContact}
        administrativeContact.selectedContacts.push(foundContact[0]);
        let filteredAvailableContacts = administrativeContact.availableContacts.filter(availableContact => {
            return foundContact[0].contactUuid !== availableContact.contactUuid
        })
        administrativeContact.availableContacts = [...filteredAvailableContacts]

        this.setState({
            administrativeContact,
        }, () => {
            this.props.editSiteContacts(false)
            this.props.siteContactData({administrativeContact: this.state.administrativeContact}, true);
            this.props.updateEditableStatus(true)
        })
    }

    updateContact = (selectedContact) => {
        let administrativeContact = { ...this.state.administrativeContact }
        let removedContact = this.getContact(selectedContact.contactUuid);
        let updatedContactTypes = administrativeContact.selectedContacts.filter(contact=>{
           return  contact.contactUuid != selectedContact.contactUuid
        })
        administrativeContact.selectedContacts = updatedContactTypes
        administrativeContact.availableContacts.push(removedContact[0])
        
        this.setState({
            administrativeContact
        }, () => {
            this.props.siteContactData({administrativeContact: this.state.administrativeContact}, true)
        }
        );
    }


    render() {
        const {
            iseditSiteContactsEnable
        } = this.props
        const {
            administrativeContact
        } = this.state;
        return (
            <Fragment>                
                {
                    <StyleGrayWrapper>
                        <CardTopContainer>
                            <Label dark={true} emphasized={true}>Administrative Contact</Label>
                            <Button disabled={iseditSiteContactsEnable} variant="secondary">
                                {i18Data.view}
                            </Button>
                        </CardTopContainer>
                        <CardBottomContainer>
                            {
                                (administrativeContact.selectedContacts && administrativeContact.selectedContacts.length> 0) &&
                                <>
                                    {administrativeContact.selectedContacts.map((contact)=>{
                                      
                                       return <>
                                        <StyledLeftSection spaceBottom>
                                        <Label dark={true} emphasized={true} >{!iseditSiteContactsEnable ? i18Data.contactName : i18Data.currentContact}</Label>
                                        <Label className='contact-name'>
                                            {contact && `${contact.firstName},${contact.lastName}`}
                                            {iseditSiteContactsEnable && <IconStyle onClick={() => this.updateContact(contact)}><Icon  className='close' graphic="close" size="xs" /></IconStyle>}
                                        </Label>
                                        </StyledLeftSection>
                                        {!iseditSiteContactsEnable &&
                                            <StyledRightSection>
                                                <Label dark={true} emphasized={true}>{i18Data.status}</Label>
                                                <Label>
                                                    {
                                                        <StatusIndicator variant={contact && contact.contactStatus === i18Data.active ? "success" : "error"}>
                                                            {contact && contact.contactStatus || i18Data.inActive}
                                                        </StatusIndicator>
                                                    }
                                                </Label>
                                            </StyledRightSection>
                                        }
                                        </>
                                    })}                                                                        
                                </>
                            }
                            {
                                (administrativeContact.availableContacts.length>0)&&
                                <DropdownWrapper>
                                    <Label dark={true} emphasized={true}>
                                        {iseditSiteContactsEnable ? 'Assign a different contact' : 'Assign contact'}
                                    </Label>
                                    <Dropdown
                                        className = 'drop-down'
                                        onChange={(e) => this.onChange(e)}
                                        defaultValue='select'
                                        value={'select'}
                                    >
                                        <option value="select">Select contact</option>
                                        {administrativeContact.availableContacts && administrativeContact.availableContacts.map((contact, index) => {
                                            return (
                                                <Fragment key={index}>
                                                    <option value={contact.contactUuid} id={contact.contactUuid} index={index}>
                                                        {`${contact.firstName},${contact.lastName}`}
                                                    </option>
                                                </Fragment>
                                            )
                                        })}
                                    </Dropdown>
                                </DropdownWrapper>
                            }

                        </CardBottomContainer>
                    </StyleGrayWrapper>
                }

            </Fragment>)
    }
}
