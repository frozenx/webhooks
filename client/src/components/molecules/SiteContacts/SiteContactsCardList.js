import React, { Fragment } from 'react';
import { FormHeading, FormHeadingWrapper, StyleGrayWrapper } from "../../../common/styles";
import Label from '@beans/label';
import Button from '@beans/button';
import i18Data from '../../../languagepack';
import AdministrativeContactCard from './AdminstrativeContactCard';

import {
    CardTopContainer,
    CardBottomContainer,
    StyledLeftSection,
    StyledRightSection,
    IconStyle
} from '../../../common/styles'
import StatusIndicator from '@beans/colleague-status-indicator';
import Dropdown from '@beans/dropdown';
import Icon from '@beans/icon';
import { DropdownWrapper } from './styled'


export default class SiteContactsCardList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            contactTypes: {
                billToContact: {},
                shipToContact: {},
                correspondenceContact: {},
                dunningContact: {},
                InvoicesContact: {}
            },
            isContactTypesPresent: false,
            editContact: false,
            editContactStatus: false,
            showLoader: false
        }
    }

    getContact = (uuid) => {                
        let foundContact = this.props.rootNode.contactBook.getContact(uuid)            
        if(foundContact) {
            let { contactStatus, firstName, lastName, contactUuid } = foundContact;
            let contact = { contactStatus, firstName, lastName, contactUuid }
            return contact
        }
        return {};
    }

    componentDidMount() {
        let allContactTypes = this.props.node.contact.values
        let savedContactTypes = {...allContactTypes}
        delete savedContactTypes.administrativeContact;
        let contactTypes = { ...this.state.contactTypes }
        let isSavedContactTypesEmpty = Object.keys(allContactTypes).every((contactType) => !allContactTypes[contactType])                
        if (!isSavedContactTypesEmpty) {
            Object.keys(savedContactTypes).map((contact, i) => {
                if (savedContactTypes[contact]) {
                    contactTypes[contact] = this.getContact(savedContactTypes[contact]);
                }
                this.setState({
                    contactTypes,
                    isContactTypesPresent: true
                })
            })
              
        }

        else {           
            this.setState({
                isContactTypesPresent: false
            })

        }            
    }
    onChange = (contactType, e) => {
        let foundContact = this.getContact(e.target.value)
        let contactTypes = {...this.state.contactTypes}
        contactTypes[contactType.id] = foundContact;
        this.setState({
            contactTypes,
            isContactTypesPresent: true
        }, () => {
            this.props.editSiteContacts(false)
            this.props.siteContactData(this.state.contactTypes, false);
        })
    }

    updateContact = (contactType) => {
        let contactTypes = { ...this.state.contactTypes }       
        contactTypes[contactType.id] = {}
        this.setState({
            contactTypes
        }, () => {
            this.props.siteContactData(this.state.contactTypes, false)
        }
        );
    }
    updateEditableStatus = (status) =>{        
        this.setState({
            isContactTypesPresent: status
        })
    }

    render() {
        const {
            contacts,
            contactList,
            rootNode,
            node,
            editSiteContacts,
            iseditSiteContactsEnable,
            siteContactData
        } = this.props
        const {
            isContactTypesPresent,
            contactTypes
        } = this.state;
        return (
            
            <Fragment>
                <FormHeadingWrapper>
                    <FormHeading><h1>{`${rootNode.partner.values.partnerName} site contacts`}</h1></FormHeading>
                    <Button className='edit-btn' disabled={iseditSiteContactsEnable || !isContactTypesPresent} onClick={() => editSiteContacts(true)} variant='secondary'>Edit site contacts </Button>
                </FormHeadingWrapper>
                <AdministrativeContactCard 
                            iseditSiteContactsEnable={iseditSiteContactsEnable}
                            editSiteContacts = {editSiteContacts}
                            contacts ={contacts}
                            rootNode = {rootNode}
                            node = {node}
                            siteContactData = {siteContactData}
                            updateEditableStatus = {(status) => this.updateEditableStatus(status)}
                        />
                {contactList.map((c, index) => {
                    let contactType = this.state.contactTypes[c.id];
                    return (
                        <>                        
                        <StyleGrayWrapper key={index}>
                        <CardTopContainer>
                            <Label dark={true} emphasized={true}>{c.label}</Label>
                            <Button disabled={iseditSiteContactsEnable} variant="secondary">
                                {i18Data.view}
                            </Button>
                        </CardTopContainer>
                        <CardBottomContainer>
                            {
                                (contactType && contactType.contactUuid) &&
                                <>                                      
                                    <StyledLeftSection spaceBottom>
                                    <Label dark={true} emphasized={true} >{!iseditSiteContactsEnable ? i18Data.contactName : i18Data.currentContact}</Label>
                                    <Label className='contact-name'>
                                        {contactType && `${contactType.firstName},${contactType.lastName}`}
                                        {iseditSiteContactsEnable && <IconStyle onClick={() => this.updateContact(c)}><Icon  className='close'  graphic="close" size="xs"/></IconStyle>}
                                    </Label>
                                    </StyledLeftSection>
                                    {!iseditSiteContactsEnable &&
                                        <StyledRightSection>
                                            <Label dark={true} emphasized={true}>{i18Data.status}</Label>
                                            <Label>
                                                {
                                                    <StatusIndicator variant={contactType && contactType.contactStatus === i18Data.active ? "success" : "error"}>
                                                        {contactType && contactType.contactStatus || i18Data.inActive}
                                                    </StatusIndicator>
                                                }
                                            </Label>
                                        </StyledRightSection>
                                    }
                                                                                                            
                                </>
                            }
                            {
                                (contactType && !contactType.contactUuid || iseditSiteContactsEnable)&&
                                <DropdownWrapper>
                                    <Label dark={true} emphasized={true}>
                                        {iseditSiteContactsEnable ? 'Assign a different contact' : 'Assign contact'}
                                    </Label>
                                    <Dropdown
                                        className='drop-down'
                                        onChange={(e) => this.onChange(c, e)}
                                        defaultValue='select'
                                        id={index}
                                        value={contactType.uuid || 'select'}
                                    >
                                        <option value="select">Select contact</option>
                                        {contacts && contacts.map((contact, index) => {
                                            return (
                                                <Fragment key={index}>
                                                    <option value={contact.contactUuid} id={contact.contactUuid} index={index}>
                                                        {`${contact.firstName},${contact.lastName}`}
                                                    </option>
                                                </Fragment>
                                            )
                                        })}
                                    </Dropdown>
                                </DropdownWrapper>
                            }

                        </CardBottomContainer>
                    </StyleGrayWrapper>
                    </>
                    )
                }

                )}
            </Fragment>)
    }
}
