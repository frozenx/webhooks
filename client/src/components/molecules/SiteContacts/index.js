import React,{Fragment, useState} from 'react';
import SiteContactsCardList from './SiteContactsCardList'
import BackComponent from '../SupplierForm/BackComponent'
import { SecondaryWrapper, BtnContainer, NotificationStyle, PrimaryWrapper } from "../../../common/styles";
import Button from '@beans/button';
import i18data from '../../../languagepack'
import {StyledContainer} from './styled'
import Notification from '@beans/notification';
import Loader from '../../atoms/Loader';
import config from '../../../config';
import httpClient from '../../../lib/httpClient';

 class SiteContacts extends React.Component {
    constructor(props){
         super(props);
         this.state = {
            contactsData: {},
            showLoader: false,
            errorTitle: i18data.formSubmitErrorTitle,
            isContactsError: false
         }
    }
    
    componentDidMount() {
        this.getContacts();
    }

    loaderStatus = (status) => {
        this.setState({
            showLoader: status
        })
    }
    getContacts = async () => {
        this.loaderStatus(true);
        try {
            const partnerUuid = this.props.rootNode.uuid;
            const url = config.endPoints.getContactsPerPage('full',partnerUuid);
            const contactRecords = await httpClient.get(url);
            const  newContacts = contactRecords.contacts.map(contact => ({...contact, sites:[]}))
            let newContactRecords = {...contactRecords,contacts:newContacts}
            const { contacts, contactCount } = newContactRecords;
            this.props.updateContactBook(contacts, contactCount);
            this.setState({
                contactsData: this.props.node.contact.values
            })
            this.loaderStatus(false);
        }
        catch(err) {
            this.setState({
                errorTitle: i18data.fetchContactsErrorTtile,
                isContactsError: true
            })
            this.loaderStatus(false);
        }
    }
    siteContactData = (data, isAdministrativeContact) => {        
        const siteContacts= {...this.state.contactsData}
          data && Object.keys(data).map(contactType => {
            let contacts = '';
            if(isAdministrativeContact) {
                   (data[contactType].selectedContacts.length> 0 ? data[contactType].selectedContacts.map(contact => {
                        contacts = contacts.concat(`,${contact.contactUuid}`)
                    }) :
                    siteContacts[contactType] = '')
                    siteContacts[contactType] = contacts.substring(1)
               }
               else {
                siteContacts[contactType] = data[contactType].contactUuid || ''

               }
          })
        this.setState({
            contactsData: siteContacts
        })
          
    }
    getSuccessTitle = (partnerName, currentFieldName) => {        
        return `${partnerName} ${currentFieldName} have been successfully updated`;
    }

    render() {
        const {
            selectedAttr,
            node,  
            contacts,
            handleBackComponent,
            rootNode,
            saveSiteContact,
            cancel,
            isFormLocked,
            updateContactBook,
            isFormSubmissionSuccessful,
            currentFieldName,
            editSiteContacts,
            iseditSiteContactsEnable,
            isFormSubmissionError
        } = this.props;
        const { showLoader, contactsData, errorTitle, isContactsError } = this.state;
       
        return(
            <StyledContainer>
                <BackComponent handleBackComponent={handleBackComponent} selectedAttr={selectedAttr}/>
                <PrimaryWrapper>
                    <SecondaryWrapper>
                        {
                            <NotificationStyle>
                                {isFormSubmissionSuccessful &&
                                <Notification title={this.getSuccessTitle(rootNode.partner.values.partnerName, currentFieldName)} variant={'success'} />}
                                {(isFormSubmissionError || isContactsError)&&
                                <Notification title={errorTitle} variant="error">
                                    {i18data.submitAgain}
                                </Notification>
                                }                                
                            </NotificationStyle>
                        }
                        { showLoader? <Loader/> :
                            <>
                                <SiteContactsCardList
                                    node={node}
                                    contactList={config.siteContactList} 
                                    contacts={contacts} 
                                    rootNode={rootNode}
                                    siteContactData={this.siteContactData}
                                    updateContactBook={updateContactBook}
                                    editSiteContacts={editSiteContacts}
                                    iseditSiteContactsEnable={iseditSiteContactsEnable}
                                />
                                <BtnContainer>
                                    <Button
                                        variant="secondary"
                                        className="submit-btn"
                                        id="save-as-draft"
                                        type="cancel"
                                        disabled={isFormLocked}
                                        onClick={(e) => { e.stopPropagation(); e.preventDefault(); cancel(e) }}
                                    > {i18data.cancel}
                                    </Button>
                                    <Button
                                        variant="primary"
                                        className={`submit - btn enabled`}
                                        type="submit"
                                        id='save'
                                        disabled={isFormLocked}
                                        onClick={(e) => { e.stopPropagation(); e.preventDefault(); saveSiteContact(selectedAttr.key, node, contactsData); }}
                                    > {i18data.save}
                                    </Button>     
                                </BtnContainer>
                            </>
                        }      
                        
                    </SecondaryWrapper>
                </PrimaryWrapper>
            </StyledContainer>        
        );
    }    
}

export default SiteContacts;