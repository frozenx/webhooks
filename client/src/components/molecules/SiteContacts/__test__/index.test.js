jest.mock('../../../../lib/httpClient');
import React from 'react';
import SiteContacts from '../'
import renderer from 'react-test-renderer';
import { DefaultThemeProvider } from '@beans/theme';
import {shallow} from 'enzyme';
import ContactBook from '../../../pages/SupplierHeader/contactBook'
import httpClient from '../../../../lib/httpClient'


describe('SiteContacts', () => {
    let props,
        tree,
        contactDefaultUrl = '/tradingpartner/api/contacts?partnerUuid=1234&pageNumber=full',
        mockContacts = [{
            ContactName: 'some contact',
            partnerName: 'some id'
        }]
    let contacts = [            
        {
          "contactUuid": "651751283168922",
          "contactStatus": "Active",
          "firstName": "Volkswagen",
          "lastName": "Zetta",
        },
        {
          "contactUuid": "600853397549052",
          "contactStatus": "Active",
          "firstName": "Skoda",
          "lastName": "Rapid",          
        }
      ]
    let linkedContacts = contacts.map(contact => ({...contact, sites:[]}));
    let contactsCount =  contacts.length;
    let siteData = 
        {
            "billToContact": {
              "contactStatus": "Inactive",
              "firstName": "Sikindar",
              "lastName": "Mirza",
              "contactUuid": "774477537246487"
            },
            "shipToContact": {
              "contactStatus": "Inactive",
              "firstName": "Sikindar",
              "lastName": "Mirza",
              "contactUuid": "774477537246487"
            },
            "correspondenceContact": {},
            "dunningContact": {
              "contactStatus": "Inactive",
              "firstName": "Sikindar",
              "lastName": "Mirza",
              "contactUuid": "774477537246487"
            },
            "InvoicesContact": {}
    }

    let administrativeContactData = {
        "administrativeContact": {
          "selectedContacts": [
            {
              "contactStatus": "Active",
              "firstName": "Skoda",
              "lastName": "Rapid",
              "contactUuid": "600853397549052"
            }
          ],
          "availableContacts": [
            {
              "contactStatus": "Active",
              "firstName": "Volkswagen",
              "lastName": "Zetta",
              "contactUuid": "651751283168922"
            }
          ]
        }
      }
    let expectedContactsData = {"InvoicesContact": "", "billToContact": "774477537246487", "correspondenceContact": "", "dunningContact": "774477537246487", "shipToContact": "774477537246487"}
    let expectedAdministrativeContactData = {"administrativeContact": "600853397549052"}

    beforeEach(() => {
        props = {
            cancel: jest.fn(),
            saveSiteContact: jest.fn(),
            selectedAttr: {
                key: 'contact'
            },
            updateContactBook: jest.fn(),            
            contacts: contacts,            
            
            node: {
                contact: {
                    values: {
                        "billToContact": "774477537246487",
                        "shipToContact": "",
                        "correspondenceContact": "",
                        "dunningContact": "",
                        "InvoicesContact": "",
                        "administrativeContact": "774477537246487"
                    }
                }
            },
                        
            rootNode: {
                uuid: 1234,
                contactBook: new ContactBook(contacts),                
                partner: {
                    values: {
                        partnerName: 'TEST NAME'                    
                    }
                }
            }

        }
        
        httpClient.get.mockResolvedValue({            
            contactCount: 2,
            contacts: [...contacts]            
        });
        tree = renderer.create(
            <DefaultThemeProvider>
                <SiteContacts  {...props}/>
            </DefaultThemeProvider>
        ).toJSON();
    });
    test('should render the component', () => {
        expect(tree).toMatchSnapshot();
    });
    test('should call getContacts service and updateContactBook prop on componentDidMount', () => {
        expect(httpClient.get).toHaveBeenCalledWith(contactDefaultUrl);
        expect(props.updateContactBook).toHaveBeenCalledWith(linkedContacts, contactsCount);
    });    
    test('should call the cancel method when cancel button is clicked', () => {
        const mockEvent = { stopPropagation: jest.fn(), preventDefault: jest.fn() };
        const Wrapper = shallow(<SiteContacts cancel= {props.cancel}/>);
        Wrapper.find('#save-as-draft').simulate('click', mockEvent)
        expect(props.cancel).toHaveBeenCalledWith(mockEvent)

    });
    describe('for the user interactions', () => {
        let Wrapper;
        beforeEach(() => {
            Wrapper = shallow(<SiteContacts saveSiteContact={props.saveSiteContact} selectedAttr={props.selectedAttr}/>);
        })
        test('should call the saveSiteContact method when save button is clicked', () => {
            const mockEvent = { stopPropagation: jest.fn(), preventDefault: jest.fn() };
            Wrapper.find('#save').simulate('click', mockEvent)
            expect(props.saveSiteContact).toHaveBeenCalled();
        });
        test('should call the saveSiteContact method without Administrative Contact', () => {
            let isAdministrativeContact = false
            const instance = Wrapper.instance();
            jest.spyOn(instance, 'siteContactData')
            instance.siteContactData(siteData, isAdministrativeContact)
            expect(Wrapper.state().contactsData).toEqual(expectedContactsData);
        });
        test('should call the saveSiteContact method with Administrative Contact', () => {
            let isAdministrativeContact = true
            const instance = Wrapper.instance();
            jest.spyOn(instance, 'siteContactData')
            instance.siteContactData(administrativeContactData, isAdministrativeContact)
            expect(Wrapper.state().contactsData).toEqual(expectedAdministrativeContactData);
        });
        test('should show the notification with when the form submission is sucessful', () => {
            const Wrapper = shallow(<SiteContacts saveSiteContact={props.saveSiteContact} selectedAttr={props.selectedAttr} isFormSubmissionSuccessful={true} rootNode={props.rootNode}/>);
            const instance = Wrapper.instance();
            jest.spyOn(instance, 'getSuccessTitle')
            instance.componentDidMount();
            expect(instance.getSuccessTitle).toHaveBeenCalled();
        });
    })
});