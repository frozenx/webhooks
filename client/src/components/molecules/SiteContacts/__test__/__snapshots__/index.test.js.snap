// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteContacts should render the component 1`] = `
<div
  className="sc-hMqMXs fTjBSR"
>
  <div
    className="sc-kkGfuU crrKaw"
  >
    <a
      className="styled__IconButtonLinkWithText-sc-1r4nugl-0 pnUgr"
      disabled={false}
      onClick={[Function]}
    >
      <span
        className="beans-link__text styled__TextSpan-sc-1r4nugl-3 jelKHq"
      >
        <span>
          Back to Partner overview
        </span>
      </span>
      <span
        className="beans-link__icon styled__IconButtonSpan-sc-1r4nugl-4 blqHEm"
        disabled={false}
      >
        <span
          className="styled__IconSpan-sc-1r4nugl-5 kuXyIv"
        >
          <svg
            aria-hidden={true}
            className="styled__Container-sc-1bf1p22-0 gJOjrx"
            color="currentColor"
            disabled={false}
            height={20}
            preserveAspectRatio="xMinYMax meet"
            viewBox="1 0 12 20"
            width={12}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.688 18.652L1.723 10l8.964-8.644"
              fill="none"
              stroke="currentColor"
            />
          </svg>
        </span>
      </span>
    </a>
  </div>
  <div
    className="sc-htpNat cVNxct"
  >
    <div
      className="base-components__RootElement-sc-150tvch-1 sc-bxivhb kUxGyj"
    >
      <div
        className="sc-ifAKCX IcNji"
      />
      <div
        className="sc-bdVaJa kybHWQ"
      >
        <span
          className="sc-bwzfXH eDyLnt"
        />
      </div>
    </div>
  </div>
</div>
`;
